src/signTx.c

```diff
@@ -365,10 +365,24 @@ static void signTx_handleInitAPDU(uint8_t p2, uint8_t* wireDataBuffer, size_t wi
 		VALIDATE(ctx->numCertificates <= SIGN_MAX_CERTIFICATES, ERR_INVALID_DATA);
 		VALIDATE(ctx->numWithdrawals <= SIGN_MAX_REWARD_WITHDRAWALS, ERR_INVALID_DATA);
 
-		// TODO isn't this more like a security policy?
-		if (ctx->commonTxData.isSigningPoolRegistrationAsOwner) {
+		switch (ctx->commonTxData.signTxUsecase) {
+		case SIGN_TX_USECASE_POOL_REGISTRATION_OWNER:
+			// necessary for security reasons
 			VALIDATE(ctx->numCertificates == 1, ERR_INVALID_DATA);
 			VALIDATE(ctx->numWithdrawals == 0, ERR_INVALID_DATA);
+			break;
+
+			#ifdef POOL_OPERATOR_APP
+		case SIGN_TX_USECASE_POOL_REGISTRATION_OPERATOR:
```

Comment:
imho we should enforce the same constraints for operators as for owners - any asymmetry in tx structure validation of those complementary usecases can cause unnecessary confusion and as a pool operator I'd appreciate a lot more the operation goes smoothly for everybody than being able to withdraw in the same transaction, even if I can do that completely separately